### PR TITLE
feat: added WatchLater and Playlists page

### PR DESCRIPTION
**Pull request by srv-pranjal**
# WatchLater and Playlists Page

## Description

 - added Playlist context for playlists and watchlater
 - added utilities for CRUD operation on playlists and watchlater
 - added playlist modal to create and add to playlists on home page
 - added private routes
 - added horizontalVideoCard and playlistCards
 
 ## Deploy Preview: [https://fino-world-git-watchlater-playlists-srv-pranjal.vercel.app/](https://fino-world-git-watchlater-playlists-srv-pranjal.vercel.app/)
image
image
image
image
image
image

### CODE REVIEW

**Review by Namrata-J — COMMENTED** [review]
liked how you have broken down components w.r.t their functionalities they are performing.
Functionality also works fine in your deploy preview.
Overall the code looks good to me.
Great work @srv-pranjal

File: src/index.js [hunk 1]
```diff
@@ -13,11 +18,13 @@ ReactDOM.render(
   <React.StrictMode>
     <Router>
       <AuthProvider>
-        <VideoProvider>
-          <SidebarProvider>
-            <App />
-          </SidebarProvider>
-        </VideoProvider>
+        <PlaylistProvider>
```
Comment (Namrata-J) [inline src/index.js:21]: could separate out all of the providers in different component

File: src/reducers/PlaylistReducer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export const PlaylistReducer = (playlistState, { type, payload }) => {
```
Comment (Namrata-J) [inline src/reducers/PlaylistReducer.js:1]: liked the way how you are destructuring it, this would have reduced lot of boilerplate code

**Review by Namrata-J — APPROVED** [review]

File: src/components/HorizontalVideoCard/HorizontalVideoCard.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <MdOutlineClose />
+      </div>
+      <div className="card__img-container">
+        <div className="card__overlay">
+          <FaPlayCircle />
+          <span>Play</span>
+        </div>
+        <img
+          src={`https://i.ytimg.com/vi/${_id}/maxresdefault.jpg`}
+          className="img-responsive"
```
Comment (akash-1318) [inline src/components/HorizontalVideoCard/HorizontalVideoCard.js:24]: Please be consistent with class names , you are using BEM for naming classes but at some places you are using different approach.

**Review by akash-1318 — COMMENTED** [review]
Just added a simple suggestion besides that everything looks good to me with good folder structure and clean code also very nicely used the combo of use reducer and useContext.
Good work.

File: src/index.js (hunk 1 above)
Comment (srv-pranjal) [reply src/index.js:21]: Thanks for the suggestion. Will do that.

File: src/components/HorizontalVideoCard/HorizontalVideoCard.js (hunk 3 above)
Comment (srv-pranjal) [reply src/components/HorizontalVideoCard/HorizontalVideoCard.js:24]: Thanks for the suggestion. Will make the changes while refactoring.

**Pull request merged**